### PR TITLE
feat(dlc): add TDD commit order detection to test coverage analysis

### LINKED ISSUE #101: feat(dlc): add TDD commit order detection to test coverage analysis
Part of #95

## Context

`dlc:test` detects missing test files and coverage gaps for recently changed files, but does not check *when* tests were written relative to source. CDT audit GAP 7 — "Tests after implementation. Tests conform to bugs rather than to spec." Kent Beck: "TDD is a superpower with AI agents." Tests written after source code tend to verify the implementation rather than the requirement — they pass but miss real bugs.

## Task

Add TDD commit order detection to `plugins/dlc/skills/test/SKILL.md` in Step 3 (Identify Coverage Gaps in Recent Changes), after the existing coverage gap checks.

## Acceptance Criteria

- [ ] VERIFY: For each changed source file with a corresponding test file, the commit order is checked via git log
- [ ] VERIFY: Source file committed before its test file produces a Low `tdd-order` finding
- [ ] VERIFY: All new source files having pre-dating tests produces an Info `tdd-order` positive finding
- [ ] VERIFY: Only files in the current branch diff (vs main) are analyzed — not the full repo history
- [ ] VERIFY: Graceful fallback when git history is shallow (no `--diff-filter=A` result)
- [ ] VERIFY: `bun scripts/validate-plugins.mjs` passes

### Files to Modify
- `plugins/dlc/skills/test/SKILL.md` — Add TDD order check in Step 3 (after line ~85); extend Step 4 severity mapping

### Approach
1. Read `plugins/dlc/skills/test/SKILL.md` and locate Step 3 (line ~69)
2. After the existing coverage gap checks (checking for test file existence and coverage %), add:

**TDD Commit Order Check:**

For each (source, test) pair identified in the branch diff:

```bash
# Get the earliest commit timestamp for the source file on this branch
SOURCE_ADDED=$(git log --format="%ct" --diff-filter=A -- "$SOURCE_FILE" | tail -1)
# Get the earliest commit timestamp for the test file on this branch
TEST_ADDED=$(git log --format="%ct" --diff-filter=A -- "$TEST_FILE" | tail -1)
# If source timestamp < test timestamp → source committed first → TDD likely violated
```

| Condition | Severity | Type | Message |
|-----------|----------|------|---------|
| Source file committed before its test file | **Low** | `tdd-order` | "Test for `{source}` written after source — risk of confirmation bias in test assertions" |
| All new source files have pre-dating tests | **Info** | `tdd-order` | "TDD discipline confirmed for this branch" |
| Source file has no test file | **High** | `coverage` | (already detected by existing check) |

**Shallow clone fallback**: If `git log --diff-filter=A` returns empty for a file (common in CI shallow clones), skip that pair and note: "Skipped TDD order check for `{file}` — insufficient git history."

3. Add `tdd-order` to Step 4 severity mapping
4. Run `bun scripts/validate-plugins.mjs` to verify

### Edge Cases
- Files added in the same commit: cannot determine order — skip with Info note
- Renamed/moved files: `--diff-filter=A` may not find the original add — skip gracefully
- Monorepo with multiple test directories: test file matching should use the existing mapping logic from the coverage gap check

### Patterns to Follow
Step 3 already iterates over changed files and checks for corresponding test files. The TDD order check is a natural extension of this loop — for files that *do* have tests, also check the commit order.

### In Scope
- Git log timestamp comparison for source vs test file creation
- Low severity finding for tests-after-source
- Info finding for confirmed TDD discipline
- Shallow clone graceful fallback

### Out of Scope
- Enforcing TDD workflow (blocking or failing the analysis)
- Checking test *quality* or assertion patterns
- Mutation testing integration
- Checking commit order within a single commit (same timestamp = skip)

## Dependencies

- Blocked by: none
- Blocks: none

### PR DESCRIPTION
**Pull request by rube-de**
## Summary

- Adds a `tdd-order` finding type to `dlc:test` that checks whether tests were committed before or after their source files using `git log --diff-filter=A` timestamps
- Source committed before test → **Low** advisory finding (confirmation bias risk)
- All tests pre-date source → **Info** positive finding (TDD discipline confirmed)
- Extends `REPORT-FORMAT.md` type enum with `tdd-order`

Closes #101

## Test plan

- [ ] Run `bun scripts/validate-plugins.mjs` — passes
- [ ] Verify `tdd-order` appears in `REPORT-FORMAT.md` type enum
- [ ] Verify `### TDD Commit Order Check` subsection in `SKILL.md` Step 3 covers: timestamp comparison, same-commit skip, shallow-clone skip
- [ ] Verify Step 4 severity table includes both Low and Info rows for TDD order

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a TDD Commit Order Check that evaluates commit relationships between test and source file pairs and emits per-pair and branch-level findings to reflect TDD order.
  * Added findings: "Source file committed before its test file" (Low) and "TDD discipline confirmed (all tests pre-date source or same-commit)" (Info).

* **Documentation**
  * Findings Report Format extended with a new "tdd-order" type.

### CODE REVIEW

File: plugins/dlc/skills/test/SKILL.md [hunk 1]
```diff
@@ -84,6 +84,36 @@ For each changed file:
...
+
+For each pair, compare timestamps:
+
+| Condition | Action |
+|-----------|--------|
+| Source timestamp < test timestamp | **Low** `tdd-order` finding — test written after source (confirmation bias risk) |
+| Test timestamp ≤ source timestamp | No finding — TDD discipline confirmed for this pair |
+| Both files in the same commit (equal timestamps) | Skip pair — no finding |
+| `--diff-filter=A` returns empty (shallow clone, renamed file) | Skip pair with note — insufficient history |
+
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/test/SKILL.md]: Fixed: clarified that shallow-clone / no-history skips are logged as a note to stdout (not a structured finding). The exact message format is now specified in the condition table.

File: plugins/dlc/skills/test/SKILL.md [hunk 2]
```diff
@@ -84,6 +84,36 @@ For each changed file:
...
+For each pair, first compare **commit hashes**, then timestamps:
+
+| Condition | Action |
+|-----------|--------|
+| Either file has no `--diff-filter=A` output (shallow clone, renamed/moved file, or file not added on this branch) | Skip pair — log a note to stdout (not a finding): "Skipping TDD order check for `{file}`: insufficient git history" |
+| Both files share the same add-commit hash | Skip pair — no finding (added atomically) |
+| Source add-commit timestamp < test add-commit timestamp | **Low** `tdd-order` finding — test written after source (confirmation bias risk) |
+| Test add-commit timestamp < source add-commit timestamp | No finding — TDD discipline confirmed for this pair |
+
+If **all** new source files in the branch diff have pre-dating or same-commit tests, emit a single **Info** `tdd-order` finding: "TDD discipline confirmed for this branch."
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/test/SKILL.md]: Fixed: added vacuous-truth guard — Info finding now requires at least one evaluated pair before emitting.

File: plugins/dlc/skills/test/SKILL.md [hunk 3]
```diff
@@ -99,7 +129,9 @@ Map results to the findings format from REPORT-FORMAT.md.
 | Changed files with < 80% coverage | **Medium** |
 | Missing test file for new module | **Medium** |
 | Flaky tests (pass on retry) | **Low** |
+| Source file committed before its test file | **Low** |
 | Coverage informational metrics | **Info** |
+| TDD discipline confirmed (all tests pre-date source) | **Info** |
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/test/SKILL.md]: Fixed: aligned Step 4 wording to include same-commit pairs, matching Step 3's allowance.

File: plugins/dlc/skills/test/SKILL.md [hunk 4]
```diff
@@ -84,6 +84,37 @@ For each changed file:
...
+For each (source file, test file) pair identified above, check whether the test was written *before* the source — a signal of TDD discipline:
+
+```bash
+# Get the commit hash and timestamp where the file was first added on this branch
+git log origin/main..HEAD --format="%H %ct" --diff-filter=A -- <source-file> | tail -1
+git log origin/main..HEAD --format="%H %ct" --diff-filter=A -- <test-file>   | tail -1
+```
+
+**Scope**: Only check files **added on the current branch** (the `origin/main..HEAD` range above). Files that already exist in `main` are excluded — the TDD order signal is only meaningful for new files introduced in this branch. Reuse the same file list from `git diff --name-only origin/main...HEAD`.
+
```
[the remark this answers is not in the document]
Comment (rube-de) [reply plugins/dlc/skills/test/SKILL.md]: Fixed: now computing merge-base once and using $MB..HEAD consistently for both git diff and git log, eliminating the range divergence.
